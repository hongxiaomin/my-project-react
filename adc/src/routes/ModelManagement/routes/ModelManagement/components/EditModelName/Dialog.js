import React from 'react'
import { assign } from 'lodash'
import classes from './EditModelName.scss'

type Props = {
  that: Object,
};
class Dialog extends React.Component {

  constructor(props) {
    super(props)
    this.state = {
    }
  }

  inputChange(type) {
    return (e) => {
      this.props.that.setState(assign({}, this.props.that.state, {
        [type]: e.target.value,
      }))
    }
  }

  props: Props;
  render() {
    return (
      <div className={classes['OneStep-container']}>
        <div className={classes.dialogBody}>
          <div>
            <div className={classes.filter}>
              Model Name:
              <input
                type='text'
                className={'pt-input'}
                value={this.props.that.state.modelName}
                onChange={this.inputChange('modelName')}
              />
            </div>
            <div className={classes.filter}>
              Description:
              <textarea
                style={{ width: '200px' }}
                className={'pt-input'}
                value={this.props.that.state.description}
                onChange={this.inputChange('description')}
              />
            </div>
          </div>
        </div>
      </div>
    )
  }
}

export default Dialog
