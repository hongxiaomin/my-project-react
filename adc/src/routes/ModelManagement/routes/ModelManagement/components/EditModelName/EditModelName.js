import React from 'react'
import { connect } from 'react-redux'
import { Modal } from 'antd'
import { injectIntl } from 'react-intl'
import DialogFilter from './Dialog'
import {
  updateModel,
} from '../../modules'

type Props = {
  toggleEdit: Function,
  dataDetail: Object,
  editOpen: Boolean,
  dispatch: Function,
  intl: Object,
}

export class EditModelName extends React.Component {
  constructor(props) {
    super(props)
    this.state = {
      visible: false,
      modelName: this.props.dataDetail.original.name,
      description: this.props.dataDetail.original.description,
    }
  }

  handleOk = () => {
    // 把获得的数据放到store中, 然后隐藏弹框
    const tmp = {
      modelId: this.props.dataDetail.original.id,
      modelName: this.state.modelName,
      description: this.state.description,
      actionType: 'update',
    }
    console.log('tmp', tmp)
    this.props.dispatch(updateModel(tmp))
    this.toggle()
  }

  toggle = () => {
    this.setState({
      visible: !this.state.visible,
    })
    this.props.toggleEdit(this.state.visible)
  }
  handleCancel = () => {
    this.setState({
      visible: false,
    })
    this.props.toggleEdit(this.state.visible)
  }

  props: Props

  render() {
    const { intl } = this.props
    return (
      <div>
        <Modal
          title={intl.formatMessage({ id: 'modelManagement.actionModel.editModel' })}
          visible={this.props.editOpen}
          onCancel={this.handleCancel}
          onOk={this.handleOk}
          okText={intl.formatMessage({ id: 'Save' })}
        >
          <DialogFilter that={this} />
        </Modal>
      </div>
    )
  }
}

export default injectIntl(connect(
  state => ({
    token: state.auth.token,
  }),
)(EditModelName))
