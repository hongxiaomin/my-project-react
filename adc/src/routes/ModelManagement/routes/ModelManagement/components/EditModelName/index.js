import EditModelName from './EditModelName'
export default EditModelName
