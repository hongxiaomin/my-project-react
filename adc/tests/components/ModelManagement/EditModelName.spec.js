import React from 'react'
import { shallow } from 'enzyme'
import { EditModelName } from 'routes/ModelManagement/routes/ModelManagement/components/EditModelName/EditModelName'

describe('(Component) EditModelName', () => {
  const toggleEdit = jest.fn()
  const dataDetail = { original: { id: '1', name: 'model' }}
  const editOpen = false
  const dispatch = jest.fn()
  const wrapper = shallow(
    <EditModelName
      toggleEdit={toggleEdit}
      dataDetail={dataDetail}
      editOpen={editOpen}
      dispatch={dispatch}
    />,
  )
  it('render normally', () => {
    expect(wrapper).toMatchSnapshot()
  })
})