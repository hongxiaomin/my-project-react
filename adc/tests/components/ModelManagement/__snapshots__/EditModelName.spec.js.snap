// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Component) EditModelName render normally 1`] = `
<div>
  <Modal
    confirmLoading={false}
    maskTransitionName="fade"
    okText="Save"
    onCancel={[Function]}
    onOk={[Function]}
    prefixCls="ant-modal"
    title="Edit Model"
    transitionName="zoom"
    visible={false}
    width={520}
  >
    <Dialog
      that={
        EditModelName {
          "_reactInternalInstance": ShallowComponentWrapper {
            "_calledComponentWillUnmount": false,
            "_compositeType": 0,
            "_context": Object {},
            "_currentElement": <EditModelName
              dataDetail={
                    Object {
                          "original": Object {
                            "id": "1",
                            "name": "model",
                          },
                        }
              }
              dispatch={[Function]}
              editOpen={false}
              toggleEdit={[Function]}
        />,
            "_debugID": 1,
            "_hostContainerInfo": null,
            "_hostParent": null,
            "_instance": [Circular],
            "_mountOrder": 1,
            "_pendingCallbacks": null,
            "_pendingElement": null,
            "_pendingForceUpdate": false,
            "_pendingReplaceState": false,
            "_pendingStateQueue": null,
            "_renderedComponent": NoopInternalComponent {
              "_currentElement": <div>
                <Modal
                        confirmLoading={false}
                        maskTransitionName="fade"
                        okText="Save"
                        onCancel={[Function]}
                        onOk={[Function]}
                        prefixCls="ant-modal"
                        title="Edit Model"
                        transitionName="zoom"
                        visible={false}
                        width={520}
                >
                        <Dialog
                                that={[Circular]}
                        />
                </Modal>
        </div>,
              "_debugID": 2,
              "_renderedOutput": <div>
                <Modal
                        confirmLoading={false}
                        maskTransitionName="fade"
                        okText="Save"
                        onCancel={[Function]}
                        onOk={[Function]}
                        prefixCls="ant-modal"
                        title="Edit Model"
                        transitionName="zoom"
                        visible={false}
                        width={520}
                >
                        <Dialog
                                that={[Circular]}
                        />
                </Modal>
        </div>,
            },
            "_renderedNodeType": 0,
            "_rootNodeID": 0,
            "_topLevelWrapper": null,
            "_updateBatchNumber": null,
            "_warnedAboutRefsInRender": false,
          },
          "context": Object {},
          "handleCancel": [Function],
          "handleOk": [Function],
          "props": Object {
            "dataDetail": Object {
              "original": Object {
                "id": "1",
                "name": "model",
              },
            },
            "dispatch": [Function],
            "editOpen": false,
            "toggleEdit": [Function],
          },
          "refs": Object {},
          "state": Object {
            "description": undefined,
            "modelName": "model",
            "visible": false,
          },
          "toggle": [Function],
          "updater": Object {
            "enqueueCallback": [Function],
            "enqueueCallbackInternal": [Function],
            "enqueueElementInternal": [Function],
            "enqueueForceUpdate": [Function],
            "enqueueReplaceState": [Function],
            "enqueueSetState": [Function],
            "isMounted": [Function],
            "validateCallback": [Function],
          },
        }
      }
    />
  </Modal>
</div>
`;
